src/auth/mas.py

```diff
@@ -119,9 +134,36 @@ def register_user(
         logger.error("Error registering new user: %s", exc.stderr)
         raise MASRegisterUserFailedError("Error validating MAS configuration.") from exc
 
+    provision_all_users(container=container)
     return password
 
 
+def provision_all_users(
+    container: ops.model.Container,
+) -> None:
+    """Start jobs to provision all users.
+
+    Args:
+        container: Synapse container.
+
+    Raises:
+        MASProvisionUserFailedError: when triggering the provisioning jobs fails.
+    """
+    command = [
+        MAS_EXECUTABLE_PATH,
+        "-c",
+        MAS_CONFIGURATION_PATH,
+        "manage",
+        "provision-all-users",
+    ]
+
+    try:
+        process = container.exec(command=command, working_dir=MAS_WORKING_DIR)
```

Comment:
Would it make sense to add a timeout here? Could this command possibly take too much time?

Reply:
This command simply schedule a background job and exit immediately iirc, so I don't think we need to add a timeout here.